dotCMS/src/integration-test/java/com/dotcms/contenttype/test/DotAssetAPITest.java

```diff
@@ -0,0 +1,106 @@
+
+        final Tuple2<Field, ContentType> fieldDotTextAssetContentType = this.createDotAssetContentType(host,
+                "text/*", "textDotAsset" + System.currentTimeMillis());
+
+        final Tuple2<Field, ContentType> fieldDotimageAssetContentType = this.createDotAssetContentType(host,
+                "image/*", "imageDotAsset" + System.currentTimeMillis());
+
+        final File tempTestFile = File
+                .createTempFile("fileTest_" + new Date().getTime(), ".txt");
+        FileUtils.writeStringToFile(tempTestFile, "Test hi this a test longer than ten characters");
+
+        Optional<ContentType> contentTypeOpt = Optional.empty();
+
+        final File tempMovieTestFile = new File(ConfigTestHelper.getPathToTestResource("images/testmovie.mov"));
+        contentTypeOpt = APILocator.getDotAssetAPI().tryMatch(
+                tempMovieTestFile, host, APILocator.systemUser());
+
+        assertTrue(contentTypeOpt.isPresent());
+        assertEquals(fieldDotVideoAssetContentType._2().variable(), contentTypeOpt.get().variable());
+
+        contentTypeOpt = APILocator.getDotAssetAPI().tryMatch(
+                tempTestFile, host, APILocator.systemUser());
+
+        assertTrue(contentTypeOpt.isPresent());
+        assertEquals(fieldDotTextAssetContentType._2().variable(), contentTypeOpt.get().variable());
+
+        final File tempImageTestFile = TestMediaCreator.createPNG();
+        contentTypeOpt = APILocator.getDotAssetAPI().tryMatch(
+                tempImageTestFile, host, APILocator.systemUser());
+
+        assertTrue(contentTypeOpt.isPresent());
+        assertEquals(fieldDotimageAssetContentType._2().variable(), contentTypeOpt.get().variable());
+
+        final File tempImageTestFile2 = TestMediaCreator.createJPEG();
+        contentTypeOpt = APILocator.getDotAssetAPI().tryMatch(
+                tempImageTestFile2, host, APILocator.systemUser());
+
+        assertTrue(contentTypeOpt.isPresent());
+        assertEquals(fieldDotimageAssetContentType._2().variable(), contentTypeOpt.get().variable());
+
+    }
+
+    private Tuple2<Field, ContentType> createDotAssetContentType (final Host host, final String accept, final String variable) throws DotSecurityException, DotDataException {
+
+        final FieldAPI fieldAPI = APILocator.getContentTypeFieldAPI();
+        final ContentTypeAPI contentTypeAPI = APILocator.getContentTypeAPI(APILocator.systemUser());
+        ContentType dotAssetContentType     = contentTypeAPI
+                .save(ContentTypeBuilder.builder(DotAssetContentType.class).folder(FolderAPI.SYSTEM_FOLDER)
+                        .host(host.getIdentifier()).name(variable)
+                        .owner(user.getUserId()).build());
+        final Map<String, Field> fieldMap = dotAssetContentType.fieldMap();
+        com.dotcms.contenttype.model.field.Field binaryField           = fieldMap.get(DotAssetContentType.ASSET_FIELD_VAR);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'com.dotcms.contenttype.model.field.Field' due to existing import 'com.dotcms.contenttype.model.field.Field'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028307)